crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -2808,13 +2808,94 @@ impl Analyzer<'_, '_> {
+                                cause: vec![],
+                            }
+                            .into());
+                        }
+                    }
+                }
+            }
+            IntrinsicKind::Lowercase => {
+                if let Type::Lit(LitType {
+                    lit: RTsLit::Str(value), ..
+                }) = r
+                {
+                    if let Some(value) = &value.raw {
+                        if !value.to_lowercase().eq(&value.to_string()) {
+                            return Err(ErrorKind::AssignFailed {
+                                span: r.span(),
+                                left: box Type::StringMapping(to.clone()),
+                                right_ident: None,
+                                right: box r.clone(),
+                                cause: vec![],
+                            }
+                            .into());
+                        }
+                    }
+                }
+            }
+            IntrinsicKind::Capitalize => {
+                if let Type::Lit(LitType {
+                    lit: RTsLit::Str(value), ..
+                }) = r
+                {
+                    if let Some(value) = &value.raw {
+                        let ch = value.to_string().chars().next();
+
+                        if let Some(ch) = ch {
+                            if !ch.is_uppercase() {
+                                return Err(ErrorKind::AssignFailed {
+                                    span: r.span(),
+                                    left: box Type::StringMapping(to.clone()),
+                                    right_ident: None,
+                                    right: box r.clone(),
+                                    cause: vec![],
+                                }
+                                .into());
+                            }
+                        }
+                    }
+                }
+            }
+            IntrinsicKind::Uncapitalize => {
+                if let Type::Lit(LitType {
+                    lit: RTsLit::Str(value), ..
+                }) = r
+                {
+                    if let Some(value) = &value.raw {
+                        let ch = value.to_string().chars().next();
+
+                        if let Some(ch) = ch {
+                            if !ch.is_lowercase() {
+                                return Err(ErrorKind::AssignFailed {
+                                    span: r.span(),
+                                    left: box Type::StringMapping(to.clone()),
+                                    right_ident: None,
+                                    right: box r.clone(),
+                                    cause: vec![],
+                                }
+                                .into());
+                            }
+                        }
+                    }
+                }
+            }
```

Comment:
Generally I think the code is overly verbose for what's its doing, any suggestions?

Reply:
code will likely change so nvm this